kernel/network/e1000.c

```diff
@@ -3,54 +3,58 @@
 #include <stdio.h>
 #include "kernel/pci/pci.h"
 #include "kernel/pci/pci_mmio.h"
+#include "kernel/mmio.h"
+#include "kernel/logger/logger.h"
+#include "kernel/memory.h"
 
-e1000_device_t e1000 = {0};
+// get some memory for our e1000
+e1000_device_t* e1000 = (e1000_device_t*) 0x5000000;
+uint32_t RX_DESC_BASE = 0x6000000;
 
 void
-writeCommand(uint16_t address, uint32_t value)
+write_command(uint16_t address, uint32_t value)
 {
-  pci_write32(e1000.mem_base+address, value);
+  pci_write32(e1000->mem_base+address, value);
 }
 
 uint32_t
-readCommand(uint16_t address)
+read_command(uint32_t address)
 {
-  return pci_read32(e1000.mem_base+address);
+  return pci_read32(e1000->mem_base+address);
 }
 
 void
 detect_eeprom()
 {
-  // write 1 to REG_EEPROM
-  writeCommand(REG_EEPROM, 0x1);
+  write_command(REG_EEPROM, 0x1);
 
   uint32_t val = 0;
-  for(uint16_t i = 0; i < 1000 && ! e1000.eeprom_exists; i++)
+  for(uint16_t i = 0; i < 1000 && ! e1000->eeprom_exists; i++)
     {
-      val = readCommand(REG_EEPROM);
+      val = read_command(REG_EEPROM);
       if(val & 0x10)
-        e1000.eeprom_exists = 1;
+        e1000->eeprom_exists = 1;
       else
-        e1000.eeprom_exists = 0;
+        e1000->eeprom_exists = 0;
     }
 #ifdef E1000_DEBUG
```

Comment:
We should remove these debug makro

Reply:
not sure yet, we might actually use this more often for debug statements like the one in the `send_packet` method that definitely does not have to be present in normal use.
But I'm not sure if the logger is capable of settings different logging levels via macro and we should use that instead.

Reply:
Not yet, but I like the idea. Maybe we should create a new issue for that. That's a bigger feature for the logger.